services/tx/api.go

```diff
@@ -307,22 +308,26 @@ func (api *PublicTxAPI) GetRawTransaction(txHash hash.Hash, verbose bool) (inter
 			return hexStr, nil
 		}
 
-		mtx = tx.Transaction()
+		mtx = tx
 	}
+	txsvalid := true
 	coinbaseAmout := uint64(0)
 	if blkHash != nil {
 		blkHashStr = blkHash.String()
-		confirmations = int64(api.txManager.bm.GetChain().BlockDAG().GetConfirmations(
-			api.txManager.bm.GetChain().BlockIndex().GetDAGBlockID(blkHash)))
+		ib := api.txManager.bm.GetChain().BlockDAG().GetBlock(blkHash)
+		if ib != nil {
+			confirmations = int64(api.txManager.bm.GetChain().BlockDAG().GetConfirmations(ib.GetID()))
+			txsvalid = (ib.GetStatus()&4 == 0)
```

Comment:
never use the magic number operation like this one. The status must be numerated by a strong type. and the masking operation must be encapsulated into a meaningful function. The magic number 4 must be eliminated.

Reply:
https://github.com/Qitmeer/qitmeer/pull/339/commits/5b4c59afa097f0b310fc529eb6ead95eaf029d6b